test/spells/effects/HealTest.cpp

```diff
@@ -372,7 +371,7 @@ TEST_P(HealApplyTest, Heals)
 
 	EXPECT_CALL(*battleFake, setUnitState(Eq(unitId), _, Gt(0))).Times(1);
 
-	EXPECT_CALL(actualCaster, getCasterUnitId()).WillRepeatedly(Return(-1));
+	EXPECT_CALL(actualCaster, getBattleCasterID()).WillRepeatedly(Return(unitId));
```

Comment:
Interesting change. So this is a test when one stack heals another stack

Reply:
It was always.

Your controversal commit is moved outside of this PR.